Author: Barry deFreese <bdefreese@debian.org>
Description: To build wise2 with the Glib version 2.0 instead of 1.2.
Forwarded: birney@sanger.ac.uk
--- wise-2.4.1.orig/src/makefile
+++ wise-2.4.1/src/makefile
@@ -31,13 +31,13 @@
 
 
 # Intel icc flags
-# CFLAGS = -c  -O3 -axWK -ipo `glib-config --cflags`
+# CFLAGS += -c  -axWK -ipo `pkg-config --cflags glib-2.0`
 
 # alpha flags (-pthread)
-# CFLAGS = -c -O3 -pthread `glib-config --cflags`
+# CFLAGS += -c -pthread `pkg-config --cflags glib-2.0`
 
 # normal linux/bsd/mac flags
-CFLAGS = -c -O3  `glib-config --cflags`
+CFLAGS += $(CPPFLAGS) -c  `pkg-config --cflags glib-2.0`
 
 
 EXTRALIBS = -lm
--- wise-2.4.1.orig/src/snp/makefile
+++ wise-2.4.1/src/snp/makefile
@@ -5,9 +5,9 @@
 CC = cc
 #CFLAGS = -c -O2  -pg -I../base/ -I../dynlibsrc/
 #CFLAGS = -c -O2 -DPTHREAD -DHAS_PTHREAD_SETSCOPE  -DUNIX  -I../base/ -I../dynlibsrc/ -I.
-CFLAGS = -c -g3 -O2  -DUNIX  -I../base/ -I../dynlibsrc/ -I. -pthread `glib-config --cflags`
+CFLAGS += $(CPPFLAGS) -c -g3  -DUNIX  -I../base/ -I../dynlibsrc/ -I. -pthread `pkg-config --cflags glib-2.0`
 
-LFLAGS =  -g -L../base/ -L../dynlibsrc/ -lpthread `glib-config --libs` -lpthread -lm
+LDFLAGS +=  -g -L../base/ -L../dynlibsrc/ -lpthread `pkg-config --libs glib-2.0` -lpthread -lm
 AR_OPTIONS = ru
 
 RANLIB_NEEDED = 0
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) $(INCFLAGS) $?
 
 test_est : test_est.o genotype.o person.o locus_framework.o locus_model_estimators.o frequency_count.o 
-	$(CC) -o test_est test_est.o genotype.o person.o locus_framework.o locus_model_estimators.o frequency_count.o ../dynlibsrc/libdyna.a ../base/libwisebase.a $(LFLAGS)
+	$(CC) -o test_est test_est.o genotype.o person.o locus_framework.o locus_model_estimators.o frequency_count.o ../dynlibsrc/libdyna.a ../base/libwisebase.a $(LDFLAGS)
 
 INCFLAGS = -I../base/ -I../dynlibsrc/
 DFLAGS = -l -D -n Wise2_ -a _api.h -b _api.t -pthreads -dbtrace 5 -nocwarn
--- wise-2.4.1.orig/src/corba/makefile
+++ wise-2.4.1/src/corba/makefile
@@ -3,9 +3,9 @@
 CC = cc
 INCFLAGS = -I../base/ -I../dynlibsrc
 #gCFLAGS = -Wall -g -pedantic -c  -DUNIX -I../base/
-#CFLAGS =  -O2 -c  -DUNIX  -I../base/ `glib-config --cflags` -I/usr/local/include/orbit-1.0/
-CFLAGS =  -pthread -c  -DUNIX  -I../base/ -I../dynlibsrc `orbit-config --cflags client`
-LFLAGS =   -L../base/ -lm -lpthread
+#CFLAGS +=  -c  -DUNIX  -I../base/ `pkg-config --cflags glib-2.0` -I/usr/local/include/orbit-1.0/
+CFLAGS  += $(CPPFLAGS) -pthread -c  -DUNIX  -I../base/ -I../dynlibsrc `orbit-config --cflags client`
+LDFLAGS += -L../base/ -lm -lpthread
 AR_OPTIONS = ru
 
 
@@ -14,18 +14,18 @@
 	orbit-idl hspscan_corba.idl
 
 test_server : test_server.o hspscan_server_impl.o hspscan_corba-common.o hspscan_corba-skels.o hspscan_corba-stubs.o corba_singleton.o 
-	cc -g -o test_server  test_server.o hspscan_server_impl.o  hspscan_corba-common.o hspscan_corba-skels.o hspscan_corba-stubs.o  corba_singleton.o ../dynlibsrc/libdyna.a ../base/libwisebase.a ../dynlibsrc/hsp.o ../dynlibsrc/subseqhash.o ../dynlibsrc/linkedlist_lookpos.o   ../dynlibsrc/libdyna_glib.a `glib-config --libs` -lm -lORBit -lIIOP -lORBitutil -lpthread
+	cc -g -o test_server  test_server.o hspscan_server_impl.o  hspscan_corba-common.o hspscan_corba-skels.o hspscan_corba-stubs.o  corba_singleton.o ../dynlibsrc/libdyna.a ../base/libwisebase.a ../dynlibsrc/hsp.o ../dynlibsrc/subseqhash.o ../dynlibsrc/linkedlist_lookpos.o   ../dynlibsrc/libdyna_glib.a `pkg-config --libs glib-2.0` -lm -lORBit -lIIOP -lORBitutil -lpthread
 
 scanwise_protein_index : scanwise_protein_index.o hspscan_server_impl.o hspscan_corba-common.o hspscan_corba-skels.o hspscan_corba-stubs.o corba_singleton.o 
-	cc -o scanwise_protein_index  scanwise_protein_index.o hspscan_server_impl.o  hspscan_corba-common.o hspscan_corba-skels.o hspscan_corba-stubs.o  corba_singleton.o  ../dynlibsrc/libdyna_glib.a ../dynlibsrc/libdyna.a ../base/libwisebase.a `glib-config --libs` -lm -lORBit -lIIOP -lORBitutil -lpthread
+	cc -o scanwise_protein_index  scanwise_protein_index.o hspscan_server_impl.o  hspscan_corba-common.o hspscan_corba-skels.o hspscan_corba-stubs.o  corba_singleton.o  ../dynlibsrc/libdyna_glib.a ../dynlibsrc/libdyna.a ../base/libwisebase.a `pkg-config --libs glib-2.0` -lm -lORBit -lIIOP -lORBitutil -lpthread
 
 
 test_client : test_client.o hspscan_corba-common.o hspscan_corba-stubs.o 
-	cc -g -o test_client  test_client.o  hspscan_corba-common.o hspscan_corba-stubs.o ../dynlibsrc/libdyna.a ../base/libwisebase.a `glib-config --libs` -lm -lORBit -lIIOP -lORBitutil -lpthread
+	cc -g -o test_client  test_client.o  hspscan_corba-common.o hspscan_corba-stubs.o ../dynlibsrc/libdyna.a ../base/libwisebase.a `pkg-config --libs glib-2.0` -lm -lORBit -lIIOP -lORBitutil -lpthread
 
 
 test_wrapper : test_wrapper.o hspscan_corba-common.o hspscan_corba-stubs.o hspscan_corba_wrapper.o corba_singleton.o 
-	cc -g -o test_wrapper test_wrapper.o  hspscan_corba-common.o hspscan_corba-stubs.o   corba_singleton.o hspscan_corba_wrapper.o ../dynlibsrc/hsp.o ../dynlibsrc/libdyna.a ../base/libwisebase.a  `glib-config --libs` -lm -lORBit -lIIOP -lORBitutil -lpthread
+	cc -g -o test_wrapper test_wrapper.o  hspscan_corba-common.o hspscan_corba-stubs.o   corba_singleton.o hspscan_corba_wrapper.o ../dynlibsrc/hsp.o ../dynlibsrc/libdyna.a ../base/libwisebase.a  `pkg-config --libs glib-2.0` -lm -lORBit -lIIOP -lORBitutil -lpthread
 
 .c.o :
 	$(CC) $(CFLAGS) $(INCFLAGS) $?
--- wise-2.4.1.orig/src/models/makefile
+++ wise-2.4.1/src/models/makefile
@@ -62,46 +62,46 @@
 #	$(CC) $(CFLAGS) -I../HMMer2/ $(INCFLAGS) wise2xhmmer2.c
 
 testgenestat : testgenestat.o genestats.o pwmdna.o 
-	$(CC) -o testgenestat testgenestat.o genestats.o pwmdna.o -ldyna -lwisebase $(LFLAGS) 
+	$(CC) -o testgenestat testgenestat.o genestats.o pwmdna.o -ldyna -lwisebase $(LDFLAGS) 
 
 test_aligng : test_aligng.o aligngenemodel.o pwmdna.o 
-	$(CC) -o test_aligng test_aligng.o aligngenemodel.o pwmdna.o  -ldyna -lwisebase $(LFLAGS) 
+	$(CC) -o test_aligng test_aligng.o aligngenemodel.o pwmdna.o  -ldyna -lwisebase $(LDFLAGS) 
 
 makepwm : makepwm.o pwmdna.o 
-	$(CC) -o makepwm makepwm.o pwmdna.o -ldyna -lwisebase $(LFLAGS) 
+	$(CC) -o makepwm makepwm.o pwmdna.o -ldyna -lwisebase $(LDFLAGS) 
 
 testwise2xhmmer2.o : testwise2xhmmer2.c
 	$(CC) $(CFLAGS) -I../HMMer2/ $(INCFLAGS) testwise2xhmmer2.c
 
 testwise2xhmmer2 : testwise2xhmmer2.o threestatemodel.o threestatedb.o pfamhmmer1db.o 
-	$(CC) -o testwise2xhmmer2 wise2xhmmer2.o testwise2xhmmer2.o threestatemodel.o threestatedb.o pfamhmmer1db.o -ldyna -lwisebase -lhmmer $(LFLAGS) 
+	$(CC) -o testwise2xhmmer2 wise2xhmmer2.o testwise2xhmmer2.o threestatemodel.o threestatedb.o pfamhmmer1db.o -ldyna -lwisebase -lhmmer $(LDFLAGS) 
 
 plan7test.o : plan7test.c
 	$(CC) $(CFLAGS) -I../HMMer2/ plan7test.c
 
 plan7test : plan7test.o plan7.o wise2xhmmer2.o threestatemodel.o threestatedb.o  pfamhmmer1db.o
-	$(CC) -o plan7test plan7test.o plan7.o threestatemodel.o threestatedb.o pfamhmmer1db.o wise2xhmmer2.o -ldyna -lwisebase -lhmmer $(LFLAGS) 
+	$(CC) -o plan7test plan7test.o plan7.o threestatemodel.o threestatedb.o pfamhmmer1db.o wise2xhmmer2.o -ldyna -lwisebase -lhmmer $(LDFLAGS) 
 
 seqstat : seqstat.o
-	$(CC) -o seqstat seqstat.o -ldyna -lwisebase $(LFLAGS)
+	$(CC) -o seqstat seqstat.o -ldyna -lwisebase $(LDFLAGS)
 
 mutatedna : mutatedna.o
-	$(CC) -o mutatedna mutatedna.o -ldyna -lwisebase $(LFLAGS) 
+	$(CC) -o mutatedna mutatedna.o -ldyna -lwisebase $(LDFLAGS) 
 
 revcomp : revcomp.o
-	$(CC) -o revcomp revcomp.o -ldyna -lwisebase $(LFLAGS) 
+	$(CC) -o revcomp revcomp.o -ldyna -lwisebase $(LDFLAGS) 
 
 stupid_motif_scan : stupid_motif_scan.o
-	$(CC) -o stupid_motif_scan stupid_motif_scan.o -ldyna -lwisebase $(LFLAGS) 
+	$(CC) -o stupid_motif_scan stupid_motif_scan.o -ldyna -lwisebase $(LDFLAGS) 
 
 seqtrunc : seqtrunc.o
-	$(CC) -o seqtrunc seqtrunc.o -ldyna -lwisebase $(LFLAGS) 
+	$(CC) -o seqtrunc seqtrunc.o -ldyna -lwisebase $(LDFLAGS) 
 
 seqlist : seqlist.o
-	$(CC) -o seqlist seqlist.o -ldyna -lwisebase $(LFLAGS) 
+	$(CC) -o seqlist seqlist.o -ldyna -lwisebase $(LDFLAGS) 
 
 testpwm : testpwm.o pwmdna.o
-	$(CC) -o testpwm testpwm.o pwmdna.o -ldyna -lwisebase $(LFLAGS)  -lefence
+	$(CC) -o testpwm testpwm.o pwmdna.o -ldyna -lwisebase $(LDFLAGS)  -lefence
 
 
 threestate : $(MODELOBJ)
@@ -111,29 +111,29 @@
 pwise : $(MODELOBJ) threestatemat.o threestatedisplay.o
 
 test : test.o proteinsw.o protprotdis.o
-	$(CC) -o test test.o proteinsw.o protprotdis.o  -ldyna_glib -ldyna -lwisebase ../dynlibsrc/matrix.o $(LFLAGS)
+	$(CC) -o test test.o proteinsw.o protprotdis.o  -ldyna_glib -ldyna -lwisebase ../dynlibsrc/matrix.o $(LDFLAGS)
 
 testanc : testanc.o proteinsw.o abc.o pba.o sw_wrap.o
-	$(CC) -o testanc testanc.o proteinsw.o abc.o pba.o sw_wrap.o $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -o testanc testanc.o proteinsw.o abc.o pba.o sw_wrap.o $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 test_tf : test_transfactor.o transfactor.o pwmdna.o
-	$(CC) -o test_tf test_transfactor.o transfactor.o pwmdna.o $(LFLAGS) -ldyna -lwisebase 
+	$(CC) -o test_tf test_transfactor.o transfactor.o pwmdna.o $(LDFLAGS) -ldyna -lwisebase 
 
 motifwise : motifwise.o transfactor.o transregion.o pwmdna.o transregiondp.o 
-	$(CC) -o motifwise motifwise.o transfactor.o transregion.o pwmdna.o transregiondp.o $(LFLAGS) -ldyna -lwisebase 
+	$(CC) -o motifwise motifwise.o transfactor.o transregion.o pwmdna.o transregiondp.o $(LDFLAGS) -ldyna -lwisebase 
 
 motifcluster : motifcluster.o transfactor.o transregion.o pwmdna.o transregiondp.o 
-	$(CC) -o motifcluster motifcluster.o transfactor.o transregion.o pwmdna.o transregiondp.o $(LFLAGS) -ldyna -lwisebase 
+	$(CC) -o motifcluster motifcluster.o transfactor.o transregion.o pwmdna.o transregiondp.o $(LDFLAGS) -ldyna -lwisebase 
 
 motifdiff : motifdiff.o transfactor.o transregion.o pwmdna.o transregiondp.o 
-	$(CC) -o motifdiff motifdiff.o transfactor.o transregion.o pwmdna.o transregiondp.o $(LFLAGS) -ldyna -lwisebase 
+	$(CC) -o motifdiff motifdiff.o transfactor.o transregion.o pwmdna.o transregiondp.o $(LDFLAGS) -ldyna -lwisebase 
 
 
 cdtest : cdtest.o cdna2genomic.o d2d_display.o
-	$(CC) -o cdtest cdtest.o cdna2genomic.o d2d_display.o  -ldyna_glib -ldyna -lwisebase $(LFLAGS)
+	$(CC) -o cdtest cdtest.o cdna2genomic.o d2d_display.o  -ldyna_glib -ldyna -lwisebase $(LDFLAGS)
 
 prob2bits : prob2bits.o
-	$(CC) -o prob2bits  $(LFLAGS) -ldyna_glib -ldyna -lwisebase 
+	$(CC) -o prob2bits  $(LDFLAGS) -ldyna_glib -ldyna -lwisebase 
 
 quick :
 	cc -O -c -I../dynlibsrc/ -I../base/ genewise6.c
@@ -180,161 +180,161 @@
 	$(CC) $(CFLAGS) fivestarscan.c -I../base/ -I../dynlibsrc/ -I../HMMer2/ -I.
 
 oldpostwise : oldpostwise.o $(NEWG) 
-	$(CC) -g -o oldpostwise oldpostwise.o $(NEWG) $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  
+	$(CC) -g -o oldpostwise oldpostwise.o $(NEWG) $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  
 
 testdna : testdna.o dnaalign.o seqaligndisplay.o 
-	$(CC) -o testdna testdna.o dnaalign.o seqaligndisplay.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -o testdna testdna.o dnaalign.o seqaligndisplay.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 dnal : dnal.o dnaalign.o seqaligndisplay.o 
-	$(CC) -o dnal dnal.o dnaalign.o seqaligndisplay.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -o dnal dnal.o dnaalign.o seqaligndisplay.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 psw : psw.o seqaligndisplay.o proteinsw.o sw_wrap.o abc.o pba.o
-	$(CC) -o psw psw.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o   $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -o psw psw.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o   $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 scanwisep.o : scanwisep.c
-	$(CC) $(CFLAGS) $(INCFLAGS) -I../external/mott `glib-config --cflags` $?
+	$(CC) $(CFLAGS) $(INCFLAGS) -I../external/mott `pkg-config --cflags glib-2.0` $?
 
 hsp2aln_sw.o : hsp2aln_sw.c
-	$(CC) $(CFLAGS) $(INCFLAGS) `glib-config --cflags` $?
+	$(CC) $(CFLAGS) $(INCFLAGS) `pkg-config --cflags glib-2.0` $?
 
 scanwisep_mysql.o : scanwisep.c
 	$(CC)  $(CFLAGS) -DSCAN_MYSQL -I../corba -I../external/mott -I../mysql/protein_index/ -I/usr/local/mysql/include scanwisep.c 
 
 scanwisep_mysql : scanwisep_mysql.o seqaligndisplay.o proteinsw.o sw_wrap.o abc.o pba.o hsp2aln_sw.o
-	$(CC) -o scanwisep scanwisep.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o ../mysql/protein_index/mysql_protein_index.o  $(LFLAGS) -L../external/mott -L/usr/local/mysql/lib -ldyna_glib -ldyna ott -lwisebase -lmysqlclient -lm $(EXTRALIBS) -lpthread -lZ
+	$(CC) -o scanwisep scanwisep.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o ../mysql/protein_index/mysql_protein_index.o  $(LDFLAGS) -L../external/mott -L/usr/local/mysql/lib -ldyna_glib -ldyna ott -lwisebase -lmysqlclient -lm $(EXTRALIBS) $(LDFLAGS) -lpthread -lZ
 
 
 scanwisep_corba.o : scanwisep.c
 	$(CC)  $(CFLAGS) -DSCAN_CORBA -I../corba -I../external/mott `orbit-config --cflags server` scanwisep.c 
 
 scanwisep_corba : scanwisep_corba.o seqaligndisplay.o proteinsw.o sw_wrap.o abc.o pba.o hsp2aln_sw.o
-	$(CC) -o scanwisep scanwisep.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o   ../corba/hspscan_corba-common.o ../corba/hspscan_corba-stubs.o   ../corba/corba_singleton.o ../corba/hspscan_corba_wrapper.o $(LFLAGS) -L../external/mott -ldyna_glib -ldyna -lmott -lwisebase  $(EXTRALIBS) `orbit-config --libs server` -lpthread 
+	$(CC) -o scanwisep scanwisep.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o   ../corba/hspscan_corba-common.o ../corba/hspscan_corba-stubs.o   ../corba/corba_singleton.o ../corba/hspscan_corba_wrapper.o $(LDFLAGS) -L../external/mott -ldyna_glib -ldyna -lmott -lwisebase  $(EXTRALIBS) `orbit-config --libs server` -lpthread 
 
 
 scanwisep_wiseserver.o : scanwisep.c
 	$(CC)  $(CFLAGS) $(INCFLAGS)  -o scanwisep_wiseserver.o -DSCAN_WISESERVER -I../network -I../socket -I../external/mott scanwisep.c 
 
 scanwise : scanwisep_wiseserver.o seqaligndisplay.o proteinsw.o sw_wrap.o abc.o pba.o hsp2aln_sw.o
-	$(CC) -o scanwise scanwisep_wiseserver.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o   ../network/net_hspscan.o ../network/client_multihspscan.o  $(LFLAGS) -L../external/mott -L../socket -lmott -ldyna_glib -ldyna -lwisesocket -lwisebase  $(EXTRALIBS) -lpthread 
+	$(CC) -o scanwise scanwisep_wiseserver.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o   ../network/net_hspscan.o ../network/client_multihspscan.o  $(LDFLAGS) -L../external/mott -L../socket -lmott -ldyna_glib -ldyna -lwisesocket -lwisebase  $(EXTRALIBS) -lpthread 
 
 scanwisep_compress.o : scanwisep.c
 	$(CC)  $(CFLAGS) -DSCAN_COMPRESS -I../dnaindex -I../network -I../socket -I../external/mott scanwisep.c 
 
 scanwisep_compress : scanwisep_compress.o seqaligndisplay.o proteinsw.o sw_wrap.o abc.o pba.o hsp2aln_sw.o
-	$(CC) -o scanwisep scanwisep.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o   ../dnaindex/compressed_protein_index.o ../dnaindex/singleseqspace.o  ../dnaindex/kmer_direct.o  ../dnaindex/kmer_index_interface.o  $(LFLAGS) -L../external/mott -L../socket -lmott -ldyna_glib -ldyna -lwisesocket -lwisebase  $(EXTRALIBS) -lpthread 
+	$(CC) -o scanwisep scanwisep.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o   ../dnaindex/compressed_protein_index.o ../dnaindex/singleseqspace.o  ../dnaindex/kmer_direct.o  ../dnaindex/kmer_index_interface.o  $(LDFLAGS) -L../external/mott -L../socket -lmott -ldyna_glib -ldyna -lwisesocket -lwisebase  $(EXTRALIBS) -lpthread 
 
 
 
 scanwisep : scanwisep.o seqaligndisplay.o proteinsw.o sw_wrap.o abc.o pba.o hsp2aln_sw.o
-	$(CC) -o scanwisep scanwisep.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o   $(LFLAGS)  -L../external/mott -lmott -ldyna_glib -ldyna  -lwisebase  $(EXTRALIBS) `glib-config --libs` -lpthread
+	$(CC) -o scanwisep scanwisep.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o   $(LDFLAGS)  -L../external/mott -lmott -ldyna_glib -ldyna  -lwisebase  $(EXTRALIBS) `pkg-config --libs glib-2.0` -lpthread
 
 test_gwhsp : test_gwhsp.o genewisehsp.o
-	$(CC) -o test_gwhsp test_gwhsp.o genewisehsp.o $(LFLAGS) -ldyna_glib -ldyna_glib -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `glib-config --libs` -lpthread
+	$(CC) -o test_gwhsp test_gwhsp.o genewisehsp.o $(LDFLAGS) -ldyna_glib -ldyna_glib -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `pkg-config --libs glib-2.0` -lpthread
 
 localcishit.o : localcishit.c
-	$(CC) $(CFLAGS) $(INCFLAGS) `glib-config --cflags` $?
+	$(CC) $(CFLAGS) $(INCFLAGS) `pkg-config --cflags glib-2.0` $?
 
 promoterwise.o : promoterwise.c  
-	$(CC) $(CFLAGS) $(INCFLAGS) `glib-config --cflags` $?
+	$(CC) $(CFLAGS) $(INCFLAGS) `pkg-config --cflags glib-2.0` $?
 
 promoterwise : promoterwise.o localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o
-	$(CC) -g -o promoterwise promoterwise.o localdba.o  localcishit.o localcispara.o dbadisplay.o  motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o  pairwiseshortdna.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `glib-config --libs` -lpthread
+	$(CC) -g -o promoterwise promoterwise.o localdba.o  localcishit.o localcispara.o dbadisplay.o  motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o  pairwiseshortdna.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `pkg-config --libs glib-2.0` -lpthread
 
 test_deng : test_deng.o localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o dnaprofile.o dnaprofiledp.o dnaprofileengine.o
-	$(CC) -o test_deng test_deng.o  localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o dnaprofile.o dnaprofiledp.o dnaprofileengine.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `glib-config --libs` -lpthread
+	$(CC) -o test_deng test_deng.o  localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o dnaprofile.o dnaprofiledp.o dnaprofileengine.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `pkg-config --libs glib-2.0` -lpthread
 
 promotercluster : promotercluster.o localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o dnaprofile.o dnaprofiledp.o dnaprofileengine.o
-	$(CC) -o promotercluster promotercluster.o  localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o dnaprofile.o dnaprofiledp.o dnaprofileengine.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `glib-config --libs` -lpthread
+	$(CC) -o promotercluster promotercluster.o  localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o dnaprofile.o dnaprofiledp.o dnaprofileengine.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `pkg-config --libs glib-2.0` -lpthread
 
 dnawise : dnawise.o dnahmm.o dnahmmdp.o seqaligndisplay.o 
-	$(CC) -o dnawise dnawise.o dnahmm.o dnahmmdp.o  seqaligndisplay.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  -lpthread
+	$(CC) -o dnawise dnawise.o dnahmm.o dnahmmdp.o  seqaligndisplay.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  -lpthread
 
 pswdb : pswdb.o seqaligndisplay.o proteinsw.o sw_wrap.o abc.o pba.o 
-	$(CC) -o pswdb pswdb.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -o pswdb pswdb.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 efpswdb : pswdb.o seqaligndisplay.o proteinsw.o sw_wrap.o abc.o pba.o 
-	$(CC) -o pswdb pswdb.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  -lefence $(EXTRALIBS)
+	$(CC) -o pswdb pswdb.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  -lefence $(EXTRALIBS)
 
 dba : dbac.o dba.o slimdba.o bigdba.o dbadisplay.o
-	$(CC) -o dba dbac.o dba.o slimdba.o bigdba.o dbadisplay.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -o dba dbac.o dba.o slimdba.o bigdba.o dbadisplay.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 lba : lba.o largeblockdp.o seqaligndisplay.o 
-	$(CC) -o lba lba.o largeblockdp.o seqaligndisplay.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -o lba lba.o largeblockdp.o seqaligndisplay.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 
 fastdba : fastdba.o dba.o slimdba.o bigdba.o dbadisplay.o
-	$(CC) -o fastdba fastdba.o dba.o slimdba.o bigdba.o dbadisplay.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -o fastdba fastdba.o dba.o slimdba.o bigdba.o dbadisplay.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 evopairwise : evopairwise.o threestatemodel.o threestateloop.o threestatedb.o pfamhmmer1db.o wise2xhmmer2.o seqaligndisplay.o 
-	$(CC) -o evopairwise evopairwise.o threestatemodel.o threestateloop.o threestatedb.o pfamhmmer1db.o  wise2xhmmer2.o seqaligndisplay.o $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -o evopairwise evopairwise.o threestatemodel.o threestateloop.o threestatedb.o pfamhmmer1db.o  wise2xhmmer2.o seqaligndisplay.o $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 test_threestatehsp : test_threestatehsp.o threestatemodel.o threestateloop.o threestatedb.o pfamhmmer1db.o wise2xhmmer2.o seqaligndisplay.o threestatehsp.o 
-	$(CC) -o test_threestatehsp test_threestatehsp.o threestatemodel.o threestateloop.o threestatedb.o pfamhmmer1db.o  seqaligndisplay.o threestatehsp.o  wise2xhmmer2.o $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -o test_threestatehsp test_threestatehsp.o threestatemodel.o threestateloop.o threestatedb.o pfamhmmer1db.o  seqaligndisplay.o threestatehsp.o  wise2xhmmer2.o $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 makerandomdb : makerandomdb.o 
-	$(CC) -o makerandomdb makerandomdb.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -o makerandomdb makerandomdb.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 genewise : genewise.o $(NEWG) 
-	$(CC) -o genewise genewise.o $(NEWG) $(LFLAGS) -lhmmer -ldyna_glib -ldyna_glib -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -o genewise genewise.o $(NEWG) $(LDFLAGS) -lhmmer -ldyna_glib -ldyna_glib -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 cdnawise : cdnawise.o $(NEWG) cdnawise10.o 
-	$(CC) -g -o cdnawise cdnawise.o cdnawise10.o $(NEWG) $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o cdnawise cdnawise.o cdnawise10.o $(NEWG) $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 fivestar : fivestar.o fivestatemodel.o fivestate.o threestatemodel.o threestatedb.o pfamhmmer1db.o wise2xhmmer2.o seqaligndisplay.o standardout.o threestatedp.o
-	$(CC) -g -o fivestar fivestar.o fivestatemodel.o fivestate.o threestatedb.o threestatemodel.o pfamhmmer1db.o wise2xhmmer2.o seqaligndisplay.o standardout.o  threestatedp.o $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o fivestar fivestar.o fivestatemodel.o fivestate.o threestatedb.o threestatemodel.o pfamhmmer1db.o wise2xhmmer2.o seqaligndisplay.o standardout.o  threestatedp.o $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 
 fivestarsearch : fivestarsearch.o fivestatemodel.o fivestate.o threestatemodel.o threestatedb.o pfamhmmer1db.o wise2xhmmer2.o seqaligndisplay.o standardout.o threestatedp.o
-	$(CC) -g -o fivestarsearch fivestarsearch.o fivestatemodel.o fivestate.o threestatedb.o threestatemodel.o pfamhmmer1db.o wise2xhmmer2.o seqaligndisplay.o standardout.o  threestatedp.o $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o fivestarsearch fivestarsearch.o fivestatemodel.o fivestate.o threestatedb.o threestatemodel.o pfamhmmer1db.o wise2xhmmer2.o seqaligndisplay.o standardout.o  threestatedp.o $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 fivestarscan : fivestarscan.o fivestatemodel.o fivestate.o threestatemodel.o threestatedb.o pfamhmmer1db.o wise2xhmmer2.o seqaligndisplay.o standardout.o threestatedp.o
-	$(CC) -g -o fivestarscan fivestarscan.o fivestatemodel.o fivestate.o threestatedb.o threestatemodel.o pfamhmmer1db.o wise2xhmmer2.o seqaligndisplay.o standardout.o  threestatedp.o $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o fivestarscan fivestarscan.o fivestatemodel.o fivestate.o threestatedb.o threestatemodel.o pfamhmmer1db.o wise2xhmmer2.o seqaligndisplay.o standardout.o  threestatedp.o $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 
 editdist : editdist.o editdistdp.o standardout.o seqaligndisplay.o 
-	$(CC) -g -o editdist editdist.o editdistdp.o standardout.o seqaligndisplay.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase 
+	$(CC) -g -o editdist editdist.o editdistdp.o standardout.o seqaligndisplay.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase 
 
 
 efgw   : genewise.o $(NEWG) 
-	$(CC) -g -o genewise genewise.o  $(NEWG)  $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) -lefence
+	$(CC) -g -o genewise genewise.o  $(NEWG)  $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) -lefence
 
 
 genewisedb : genewisedb.o $(NEWG) 
-	$(CC) -g -o genewisedb genewisedb.o  $(NEWG)  $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) 
+	$(CC) -g -o genewisedb genewisedb.o  $(NEWG)  $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) 
 
 estwisedb : estwisedb.o $(NEWG) 
-	$(CC) -g -o estwisedb estwisedb.o $(NEWG)  $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o estwisedb estwisedb.o $(NEWG)  $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 edb : estwisedb.o $(NEWG) 
-	$(CC) -g -o estwisedb estwisedb.o $(NEWG)  $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) -lefence
+	$(CC) -g -o estwisedb estwisedb.o $(NEWG)  $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) -lefence
 
 estwise : estwise.o $(NEWG) 
-	$(CC) -g -o estwise estwise.o  $(NEWG) $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o estwise estwise.o  $(NEWG) $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 
 amplimer_resolver : amplimer_resolver.o dnaalign.o dnamatcher.o 
-	$(CC) -g -o amplimer_resolver amplimer_resolver.o dnamatcher.o dnaalign.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o amplimer_resolver amplimer_resolver.o dnamatcher.o dnaalign.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 genomewise : genomewise.o genomewise9.o genome_evidence.o est_evidence.o geneutil.o geneoutput.o
-	$(CC) -g -o genomewise genomewise.o genomewise9.o genome_evidence.o est_evidence.o geneoutput.o geneutil.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o genomewise genomewise.o genomewise9.o genome_evidence.o est_evidence.o geneoutput.o geneutil.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 sywise : sywise.o sywise20.o syexonmodel.o genestats.o pwmdna.o geneutil.o standardout.o
-	$(CC) -g -o sywise sywise.o sywise20.o syexonmodel.o genestats.o pwmdna.o  standardout.o geneutil.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o sywise sywise.o sywise20.o syexonmodel.o genestats.o pwmdna.o  standardout.o geneutil.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 alignwise : alignwise.o alignwisedp.o aligngenemodel.o genestats.o pwmdna.o geneutil.o standardout.o geneoutput.o
-	$(CC) -g -o alignwise alignwise.o alignwisedp.o aligngenemodel.o genestats.o pwmdna.o geneutil.o standardout.o geneoutput.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o alignwise alignwise.o alignwisedp.o aligngenemodel.o genestats.o pwmdna.o geneutil.o standardout.o geneoutput.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 test_splice : test_splice.o  aligngenemodel.o genestats.o pwmdna.o geneutil.o standardout.o
-	$(CC) -g -o test_splice test_splice.o aligngenemodel.o genestats.o pwmdna.o geneutil.o standardout.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o test_splice test_splice.o aligngenemodel.o genestats.o pwmdna.o geneutil.o standardout.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 statwise : statwise.o statwise10.o syexonmodel.o genestats.o pwmdna.o geneutil.o
-	$(CC) -g -o statwise statwise.o statwise10.o syexonmodel.o genestats.o pwmdna.o geneutil.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o statwise statwise.o statwise10.o syexonmodel.o genestats.o pwmdna.o geneutil.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 pseudowise : pseudowise.o pseudowise7.o $(NEWG) 
-	$(CC) -g -o pseudowise pseudowise.o pseudowise7.o $(NEWG) $(LFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o pseudowise pseudowise.o pseudowise7.o $(NEWG) $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 clonewise : clonewise.o clonewisedp.o localclonewisedp.o mapstruct.o
-	$(CC) -g -o clonewise clonewise.o clonewisedp.o localclonewisedp.o mapstruct.o $(LFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o clonewise clonewise.o clonewisedp.o localclonewisedp.o mapstruct.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 
 HMMER_LIBS = ../HMMer2/
@@ -345,11 +345,11 @@
 CC = cc
 #CFLAGS = -c -O2  -pg -I../base/ -I../dynlibsrc/
 #CFLAGS = -c -O2 -DPTHREAD -DHAS_PTHREAD_SETSCOPE  -DUNIX  -I../base/ -I../dynlibsrc/ -I.
-CFLAGS = -c -g3 -O2  -DUNIX  -I../base/ -I../dynlibsrc/ -I. -pthread `glib-config --cflags`
-#CFLAGS = -c  -g -DHAS_PTHREAD_SETSCOPE -DUNIX -I../base/ -I../dynlibsrc/ `glib-config --cflags`
-#LFLAGS =  -L../base/ -L../dynlibsrc/ -L../HMMer2/  
+CFLAGS += -c -g3  -DUNIX  -I../base/ -I../dynlibsrc/ -I. -pthread `pkg-config --cflags glib-2.0`
+#CFLAGS += -c  -g -DHAS_PTHREAD_SETSCOPE -DUNIX -I../base/ -I../dynlibsrc/ `pkg-config --cflags glib-2.0`
+#LDFLAGS =  -L../base/ -L../dynlibsrc/ -L../HMMer2/  
 
-LFLAGS =  -L../base/ -L../dynlibsrc/ -L$(HMMER_LIBS) -lpthread `glib-config --libs`
+LDFLAGS +=  -L../base/ -L../dynlibsrc/ -L$(HMMER_LIBS) -lpthread `pkg-config --libs glib-2.0`
 AR_OPTIONS = ru
 
 RANLIB_NEEDED = 0
@@ -360,7 +360,7 @@
 
 
 #test : test.o libwisedb.a 
-#	$(CC) -o test  test.o libwisedb.a  -lprob -ldyn  -lwisebase $(LFLAGS)
+#	$(CC) -o test  test.o libwisedb.a  -lprob -ldyn  -lwisebase $(LDFLAGS)
 
 
 #
--- wise-2.4.1.orig/src/dnaindex/makefile
+++ wise-2.4.1/src/dnaindex/makefile
@@ -27,53 +27,53 @@
 
 
 kwise : kwise.o $(ASSEMBLY_OBJ)
-	$(CC) -o kwise kwise.o $(ASSEMBLY_OBJ)  -ldyna -lwisebase $(LFLAGS) `glib-config --libs`
+	$(CC) -o kwise kwise.o $(ASSEMBLY_OBJ)  -ldyna -lwisebase $(LDFLAGS) `pkg-config --libs glib-2.0`
 
 arraymatcher : arraymatcher.o $(ASSEMBLY_OBJ) 
-	$(CC) -o arraymatcher arraymatcher.o $(ASSEMBLY_OBJ)  -ldyna -lwisebase $(LFLAGS) `glib-config --libs`
+	$(CC) -o arraymatcher arraymatcher.o $(ASSEMBLY_OBJ)  -ldyna -lwisebase $(LDFLAGS) `pkg-config --libs glib-2.0`
 
 
 findbad_kmer : findbad_kmer.o $(ASSEMBLY_OBJ)
-	$(CC) -o findbad_kmer findbad_kmer.o $(ASSEMBLY_OBJ)  -ldyna -lwisebase $(LFLAGS) `glib-config --libs`
+	$(CC) -o findbad_kmer findbad_kmer.o $(ASSEMBLY_OBJ)  -ldyna -lwisebase $(LDFLAGS) `pkg-config --libs glib-2.0`
 
 make_shotgun : make_shotgun.o shotgun.o
-	$(CC) -o make_shotgun make_shotgun.o shotgun.o -ldyna -lwisebase $(LFLAGS)
+	$(CC) -o make_shotgun make_shotgun.o shotgun.o -ldyna -lwisebase $(LDFLAGS)
 
 test_dnanumber : test_dnanumber.o dnanumber.o
-	$(CC) -o test_dnanumber test_dnanumber.o dnanumber.o -ldyna -lwisebase $(LFLAGS)
+	$(CC) -o test_dnanumber test_dnanumber.o dnanumber.o -ldyna -lwisebase $(LDFLAGS)
 
 test_linkindex : test_linkindex.o linkindex.o dnanumber.o
-	$(CC) -o test_linkindex test_linkindex.o linkindex.o dnanumber.o -ldyna -lwisebase $(LFLAGS)
+	$(CC) -o test_linkindex test_linkindex.o linkindex.o dnanumber.o -ldyna -lwisebase $(LDFLAGS)
 
 test_forwardindex : test_forwardindex.o forwardindex.o 
-	$(CC) -o test_forwardindex test_forwardindex.o forwardindex.o -ldyna -lwisebase $(LFLAGS)
+	$(CC) -o test_forwardindex test_forwardindex.o forwardindex.o -ldyna -lwisebase $(LDFLAGS)
 
 test_euler : test_euler.o eulerindex.o singleseqspace.o
-	$(CC) -o test_euler test_euler.o eulerindex.o singleseqspace.o -ldyna -lwisebase $(LFLAGS)
+	$(CC) -o test_euler test_euler.o eulerindex.o singleseqspace.o -ldyna -lwisebase $(LDFLAGS)
 
 test_kmer : test_kmer.o kmer_assembly.o kmer_index_interface.o kmer_direct.o singleseqspace.o dnamapping.o largeseqreader.o  kmer_assembly_untangler.o kmer_assembly_contig.o kmer_assembly_error.o assembly.o 
-	$(CC) -o test_kmer  test_kmer.o kmer_assembly.o kmer_index_interface.o kmer_direct.o singleseqspace.o dnamapping.o  largeseqreader.o kmer_assembly_untangler.o   kmer_assembly_error.o kmer_assembly_contig.o assembly.o  -ldyna -lwisebase $(LFLAGS)
+	$(CC) -o test_kmer  test_kmer.o kmer_assembly.o kmer_index_interface.o kmer_direct.o singleseqspace.o dnamapping.o  largeseqreader.o kmer_assembly_untangler.o   kmer_assembly_error.o kmer_assembly_contig.o assembly.o  -ldyna -lwisebase $(LDFLAGS)
 
 kmer_stat : kmer_stat.o kmer_assembly.o kmer_index_interface.o kmer_direct.o singleseqspace.o dnamapping.o largeseqreader.o 
-	$(CC) -o kmer_stat kmer_stat.o kmer_assembly.o kmer_index_interface.o kmer_direct.o singleseqspace.o dnamapping.o  largeseqreader.o -ldyna -lwisebase $(LFLAGS)
+	$(CC) -o kmer_stat kmer_stat.o kmer_assembly.o kmer_index_interface.o kmer_direct.o singleseqspace.o dnamapping.o  largeseqreader.o -ldyna -lwisebase $(LDFLAGS)
 
 test_compara : test_compara.o comparapath.o dnamapping.o singleseqspace.o chhash.o largeseqreader.o kmer_direct.o kmer_hash.o
-	$(CC) -o test_compara test_compara.o comparapath.o dnamapping.o singleseqspace.o chhash.o largeseqreader.o kmer_direct.o kmer_hash.o -ldyna -lwisebase $(LFLAGS)
+	$(CC) -o test_compara test_compara.o comparapath.o dnamapping.o singleseqspace.o chhash.o largeseqreader.o kmer_direct.o kmer_hash.o -ldyna -lwisebase $(LDFLAGS)
 
 bigseqwise : bigseqwise.o comparapath.o dnamapping.o singleseqspace.o chhash.o largeseqreader.o kmer_direct.o kmer_hash.o
-	$(CC) -o bigseqwise bigseqwise.o comparapath.o dnamapping.o singleseqspace.o chhash.o largeseqreader.o kmer_direct.o kmer_hash.o -ldyna -lwisebase $(LFLAGS)
+	$(CC) -o bigseqwise bigseqwise.o comparapath.o dnamapping.o singleseqspace.o chhash.o largeseqreader.o kmer_direct.o kmer_hash.o -ldyna -lwisebase $(LDFLAGS)
 
 
 
 
 show_euler : show_euler.o eulerindex.o singleseqspace.o
-	$(CC) -o show_euler show_euler.o eulerindex.o singleseqspace.o -ldyna -lwisebase $(LFLAGS)
+	$(CC) -o show_euler show_euler.o eulerindex.o singleseqspace.o -ldyna -lwisebase $(LDFLAGS)
 
 CC = cc
-CFLAGS = -pthread -c  -DHAS_PTHREAD_SETSCOPE -g -DUNIX -I../base/ -I../dynlibsrc/ `glib-config --cflags`
+CFLAGS += $(CPPFLAGS) -pthread -c  -DHAS_PTHREAD_SETSCOPE -g -DUNIX -I../base/ -I../dynlibsrc/ `pkg-config --cflags glib-2.0`
 
 
-LFLAGS =  -g -L../base/ -L../dynlibsrc/ -lpthread `glib-config --libs` -lm
+LDFLAGS +=  -g -L../base/ -L../dynlibsrc/ -lpthread `pkg-config --libs glib-2.0` -lm
 
 
 .c.o :
--- wise-2.4.1.orig/src/dnaindex/assembly/makefile
+++ wise-2.4.1/src/dnaindex/assembly/makefile
@@ -3,33 +3,33 @@
 .SUFFIXES :  .dy
 
 test_basegraph : test_basegraph.o basegraph.o
-	$(CC) -o test_basegraph test_basegraph.o basegraph.o ../dnamapping.o ../kmer_index_interface.o ../kmer_glib_index.o $(LFLAGS) -ldyna -lwisebase
+	$(CC) -o test_basegraph test_basegraph.o basegraph.o ../dnamapping.o ../kmer_index_interface.o ../kmer_glib_index.o $(LDFLAGS) -ldyna -lwisebase
 
 test_assemblygraph : test_assemblygraph.o assemblygraph.o assemblystats.o basegraph.o
-	$(CC) -o test_assemblygraph test_assemblygraph.o  assemblystats.o basegraph.o assemblygraph.o ../dnamapping.o ../kmer_index_interface.o ../kmer_glib_index.o ../kmer_hash.o ../assembly.o ../assembly_stream_fasta.o ../largeseqreader.o ../singleseqspace.o ../assembly_stream_interface.o $(LFLAGS) -ldyna -lwisebase
+	$(CC) -o test_assemblygraph test_assemblygraph.o  assemblystats.o basegraph.o assemblygraph.o ../dnamapping.o ../kmer_index_interface.o ../kmer_glib_index.o ../kmer_hash.o ../assembly.o ../assembly_stream_fasta.o ../largeseqreader.o ../singleseqspace.o ../assembly_stream_interface.o $(LDFLAGS) -ldyna -lwisebase
 
 
 test_graphtangle : test_graphtangle.o assemblygraph.o assemblystats.o basegraph.o graphtangle.o assemblypath.o 
-	$(CC) -o test_graphtangle test_graphtangle.o  assemblystats.o basegraph.o assemblygraph.o graphtangle.o assemblypath.o ../dnamapping.o ../kmer_index_interface.o ../kmer_glib_index.o ../assembly.o ../assembly_stream_fasta.o ../largeseqreader.o ../singleseqspace.o ../assembly_stream_interface.o ../kmer_hash.o $(LFLAGS) -ldyna -lwisebase
+	$(CC) -o test_graphtangle test_graphtangle.o  assemblystats.o basegraph.o assemblygraph.o graphtangle.o assemblypath.o ../dnamapping.o ../kmer_index_interface.o ../kmer_glib_index.o ../assembly.o ../assembly_stream_fasta.o ../largeseqreader.o ../singleseqspace.o ../assembly_stream_interface.o ../kmer_hash.o $(LDFLAGS) -ldyna -lwisebase
 
 
 pathwise : pathwise.o supertangle.o assemblygraph.o assemblystats.o basegraph.o graphtangle.o assemblypath.o grapherror.o graphmanager.o depthmanager.o 
-	$(CC) -o pathwise pathwise.o  supertangle.o depthmanager.o assemblystats.o basegraph.o assemblygraph.o ../dnamapping.o ../kmer_index_interface.o graphtangle.o assemblypath.o grapherror.o graphmanager.o  ../kmer_glib_index.o ../kmer_hash.o ../assembly.o ../assembly_stream_fasta.o ../largeseqreader.o ../singleseqspace.o ../assembly_stream_interface.o ../assembly_sanger_project.o $(LFLAGS) -ldyna -lwisebase
+	$(CC) -o pathwise pathwise.o  supertangle.o depthmanager.o assemblystats.o basegraph.o assemblygraph.o ../dnamapping.o ../kmer_index_interface.o graphtangle.o assemblypath.o grapherror.o graphmanager.o  ../kmer_glib_index.o ../kmer_hash.o ../assembly.o ../assembly_stream_fasta.o ../largeseqreader.o ../singleseqspace.o ../assembly_stream_interface.o ../assembly_sanger_project.o $(LDFLAGS) -ldyna -lwisebase
 
 
 badkmer : badkmer.o assemblygraph.o assemblystats.o basegraph.o
-	$(CC) -o badkmer badkmer.o  assemblystats.o basegraph.o assemblygraph.o ../dnamapping.o ../kmer_index_interface.o ../kmer_glib_index.o ../kmer_hash.o ../assembly.o ../assembly_stream_fasta.o ../largeseqreader.o ../singleseqspace.o ../assembly_stream_interface.o $(LFLAGS) -ldyna -lwisebase
+	$(CC) -o badkmer badkmer.o  assemblystats.o basegraph.o assemblygraph.o ../dnamapping.o ../kmer_index_interface.o ../kmer_glib_index.o ../kmer_hash.o ../assembly.o ../assembly_stream_fasta.o ../largeseqreader.o ../singleseqspace.o ../assembly_stream_interface.o $(LDFLAGS) -ldyna -lwisebase
 
 arraymatcher : arraymatcher.o assemblygraph.o assemblystats.o basegraph.o
-	$(CC) -o arraymatcher arraymatcher.o  assemblystats.o basegraph.o assemblygraph.o ../dnamapping.o ../kmer_index_interface.o ../kmer_glib_index.o ../kmer_hash.o ../assembly.o ../assembly_stream_fasta.o ../largeseqreader.o ../singleseqspace.o ../assembly_stream_interface.o $(LFLAGS) -ldyna -lwisebase
+	$(CC) -o arraymatcher arraymatcher.o  assemblystats.o basegraph.o assemblygraph.o ../dnamapping.o ../kmer_index_interface.o ../kmer_glib_index.o ../kmer_hash.o ../assembly.o ../assembly_stream_fasta.o ../largeseqreader.o ../singleseqspace.o ../assembly_stream_interface.o $(LDFLAGS) -ldyna -lwisebase
 
 
 
 CC = cc
-CFLAGS = -Wall -pthread -c  -DHAS_PTHREAD_SETSCOPE -g -DUNIX -I../../base/ -I../../dynlibsrc/ -I../ -I../../models/ `glib-config --cflags`
+CFLAGS += $(CPPFLAGS) -Wall -pthread -c  -DHAS_PTHREAD_SETSCOPE -g -DUNIX -I../../base/ -I../../dynlibsrc/ -I../ -I../../models/ `pkg-config --cflags glib-2.0`
 
 
-LFLAGS =  -g -L../../base/ -L../../dynlibsrc/ -lpthread `glib-config --libs` -lm
+LDFLAGS +=  -g -L../../base/ -L../../dynlibsrc/ -lpthread `pkg-config --libs glib-2.0` -lm
 
 
 .c.o :
--- wise-2.4.1.orig/src/dynlibsrc/makefile
+++ wise-2.4.1/src/dynlibsrc/makefile
@@ -84,11 +84,11 @@
 CC = cc
 INCFLAGS = -I../base/
 #CFLAGS = -Wall -g -pedantic -c  -DUNIX -I../base/
-CFLAGS =  -Wall -pthread -g3 -DCOMPILE_VERBOSITY -DPTHREAD -O2 -c  -DUNIX -I../base/ `glib-config --cflags`
-#CFLAGS =  -Wall -g3 -pg -DPTHREAD -O2  -c  -DUNIX -I../base/ `glib-config --cflags`
-#CFLAGS =   -O2 -c -pthread  -DUNIX  -I../base/ `glib-config --cflags`
+CFLAGS += $(CPPFLAGS) -Wall -pthread -g3 -DCOMPILE_VERBOSITY -DPTHREAD -c  -DUNIX -I../base/ `pkg-config --cflags glib-2.0`
+#CFLAGS +=  -Wall -g3 -pg -DPTHREAD  -c  -DUNIX -I../base/ `pkg-config --cflags glib-2.0`
+#CFLAGS +=   -c -pthread  -DUNIX  -I../base/ `pkg-config --cflags glib-2.0`
 #CFLAGS =   -O -c -DUNIX -DNOERROR -I../base/
-LFLAGS =  -g -L../base/ -lm 
+LDFLAGS +=  -g -L../base/ -lm 
 AR_OPTIONS = ru
 
 RANLIB_NEEDED = 0
@@ -133,47 +133,47 @@
 #
 
 test_genomic : genomic.o test_genomic.o sequence.o codon.o
-	$(CC) -o test_genomic test_genomic.o genomic.o sequence.o codon.o ../base/libwisebase.a -lm
+	$(CC) $(LDFLAGS)-o test_genomic test_genomic.o genomic.o sequence.o codon.o ../base/libwisebase.a -lm
 
 test_tree : tree.o test_tree.o
-	$(CC) -o test_tree test_tree.o tree.o libdyna.a ../base/libwisebase.a -lm
+	$(CC) $(LDFLAGS) -o test_tree test_tree.o tree.o libdyna.a ../base/libwisebase.a -lm
 
 test_intallocator : intallocator.o test_intallocator.o 
-	$(CC) -o test_intallocator intallocator.o test_intallocator.o  ../base/libwisebase.a -lm
+	$(CC) $(LDFLAGS) -o test_intallocator intallocator.o test_intallocator.o  ../base/libwisebase.a -lm
 
 test_psi : test_psi.o proteinstreamedindex.o 
-	$(CC) -o test_psi test_psi.o  proteinstreamedindex.o ./libdyna.a ../base/libwisebase.a -lm
+	$(CC) $(LDFLAGS) -o test_psi test_psi.o  proteinstreamedindex.o ./libdyna.a ../base/libwisebase.a -lm
 
 test_shadowindex : test_shadowindex.o shadowseqindex.o shadowseq.o libdyna.a
-	$(CC) -g -o test_shadowindex test_shadowindex.o shadowseqindex.o shadowseq.o ./libdyna.a ../base/libwisebase.a -lm
+	$(CC) $(LDFLAGS) -g -o test_shadowindex test_shadowindex.o shadowseqindex.o shadowseq.o ./libdyna.a ../base/libwisebase.a -lm
 
 test_seqhash : test_seqhash.o subseqhash.o hsp.o hitlist.o hspscaninterface.o hsplookupscan.o libdyna.a
-	$(CC) -g -o test_seqhash `glib-config --libs` test_seqhash.o subseqhash.o hsp.o  hitlist.o hspscaninterface.o hsplookupscan.o libdyna.a ../base/libwisebase.a -lm  
+	$(CC) $(LDFLAGS) -g -o test_seqhash `pkg-config --libs glib-2.0` test_seqhash.o subseqhash.o hsp.o  hitlist.o hspscaninterface.o hsplookupscan.o libdyna.a ../base/libwisebase.a -lm  
 
 test_hsp : test_hsp.o codon.o sequence.o hsp.o compmat.o probability.o 
-	$(CC) -g -o test_hsp test_hsp.o codon.o sequence.o hsp.o compmat.o probability.o  ../base/libwisebase.a -lm `glib-config --libs`
+	$(CC) $(LDFLAGS) -g -o test_hsp test_hsp.o codon.o sequence.o hsp.o compmat.o probability.o  ../base/libwisebase.a -lm `pkg-config --libs glib-2.0`
 
 test_topscore : test_topscore.o codon.o sequence.o hsp.o compmat.o probability.o 
-	$(CC) -g -o test_topscore test_topscore.o codon.o sequence.o hsp.o compmat.o probability.o  ../base/libwisebase.a -lm `glib-config --libs`
+	$(CC) $(LDFLAGS) -g -o test_topscore test_topscore.o codon.o sequence.o hsp.o compmat.o probability.o  ../base/libwisebase.a -lm `pkg-config --libs glib-2.0`
 
 
 testgendb : probability.o genomicdb.o genomic.o testgendb.o sequence.o codon.o sequencedb.o complexsequence.o complexevalset.o
-	$(CC) -o testgendb probability.o genomicdb.o genomic.o testgendb.o sequence.o codon.o sequencedb.o complexsequence.o complexevalset.o ../base/libwisebase.a -lm
+	$(CC) $(LDFLAGS) -o testgendb probability.o genomicdb.o genomic.o testgendb.o sequence.o codon.o sequencedb.o complexsequence.o complexevalset.o ../base/libwisebase.a -lm
 
 testseqalign : seqalign.o testseqalign.o sequence.o codon.o
-	$(CC) -o testseqalign testseqalign.o seqalign.o sequence.o codon.o ../base/libwisebase.a -lm
+	$(CC) $(LDFLAGS) -o testseqalign testseqalign.o seqalign.o sequence.o codon.o ../base/libwisebase.a -lm
 
 testanc : libdyna.a
-	$(CC) -o testanc ../models/testanc.o ../models/proteinsw.o ../models/abc.o ../models/pba.o ../models/sw_wrap.o libdyna.a ../base/libwisebase.a -lm
+	$(CC) $(LDFLAGS) -o testanc ../models/testanc.o ../models/proteinsw.o ../models/abc.o ../models/pba.o ../models/sw_wrap.o libdyna.a ../base/libwisebase.a -lm
 
 testdb : testdb.o sequencedb.o sequence.o codon.o probability.o
-	$(CC) -o testdb testdb.o sequencedb.o sequence.o codon.o probability.o ../base/libwisebase.a -lm
+	$(CC) $(LDFLAGS) -o testdb testdb.o sequencedb.o sequence.o codon.o probability.o ../base/libwisebase.a -lm
 
 testpdb : testpdb.o protein.o proteindb.o complexevalset.o complexsequence.o sequencedb.o sequence.o codon.o probability.o randomdb.o randommodel.o histogram.o 
-	$(CC) -o testpdb testpdb.o protein.o proteindb.o complexsequence.o sequencedb.o sequence.o codon.o probability.o randomdb.o randommodel.o complexevalset.o histogram.o ../base/libwisebase.a -lm
+	$(CC) $(LDFLAGS) -o testpdb testpdb.o protein.o proteindb.o complexsequence.o sequencedb.o sequence.o codon.o probability.o randomdb.o randommodel.o complexevalset.o histogram.o ../base/libwisebase.a -lm
 
 testpb : testpairbase.o pairbase.o pairbaseseq.o sequence.o seqalign.o codon.o probability.o complexsequence.o complexevalset.o
-	$(CC) -g -o testpb testpairbase.o pairbase.o pairbaseseq.o sequence.o codon.o seqalign.o probability.o complexsequence.o complexevalset.o ../base/libwisebase.a -lm
+	$(CC) $(LDFLAGS) -g -o testpb testpairbase.o pairbase.o pairbaseseq.o sequence.o codon.o seqalign.o probability.o complexsequence.o complexevalset.o ../base/libwisebase.a -lm
 
 .c.o :
 	$(CC) $(CFLAGS) $(INCFLAGS) $?
--- wise-2.4.1.orig/src/network/makefile
+++ wise-2.4.1/src/network/makefile
@@ -3,8 +3,8 @@
 
 CC = cc
 INCFLAGS = -I../base/  -I../socket -I../dynlibsrc -I../dnaindex
-CFLAGS =    -O2  -c -pthread  -DUNIX  -I../base/ -I../socket -I../dynlibsrc -I../dnaindex `glib-config --cflags`
-LFLAGS =  -g -L../base/ -L../socket -L../dynlibsrc -L../dnaindex -lm `glib-config --libs`
+CFLAGS += $(CPPFLAGS)  -c -pthread  -DUNIX  -I../base/ -I../socket -I../dynlibsrc -I../dnaindex `pkg-config --cflags glib-2.0`
+LDFLAGS +=  -g -L../base/ -L../socket -L../dynlibsrc -L../dnaindex -lm `pkg-config --libs glib-2.0`
 AR_OPTIONS = ru
 
 RANLIB_NEEDED = 0
@@ -13,7 +13,7 @@
 
 
 scanwise_server : wise_proteinindex_server.o net_hspscan.o ../dynlibsrc/libdyna.a ../dynlibsrc/libdyna_glib.a ../dnaindex/compressed_protein_index.o ../dnaindex/kmer_index_interface.o ../dnaindex/singleseqspace.o ../dnaindex/kmer_direct.o 
-	$(CC) -g -o scanwise_server wise_proteinindex_server.o net_hspscan.o ../dnaindex/compressed_protein_index.o ../dnaindex/kmer_index_interface.o ../dnaindex/singleseqspace.o ../dnaindex/kmer_direct.o  -ldyna_glib -ldyna -lwisesocket -lwisebase $(LFLAGS) -lpthread 
+	$(CC) -g -o scanwise_server wise_proteinindex_server.o net_hspscan.o ../dnaindex/compressed_protein_index.o ../dnaindex/kmer_index_interface.o ../dnaindex/singleseqspace.o ../dnaindex/kmer_direct.o  -ldyna_glib -ldyna -lwisesocket -lwisebase $(LDFLAGS) -lpthread 
 
 
 .c.o :
--- wise-2.4.1.orig/src/other_programs/makefile
+++ wise-2.4.1/src/other_programs/makefile
@@ -3,8 +3,8 @@
 .SUFFIXES :  .dy
 
 CC = cc
-CFLAGS =   -O2 -g3 -c -pthread  -DUNIX  -I../base/ -I../dynlibsrc `glib-config --cflags`
-LFLAGS =  -g -L../base/ -L../dynlibsrc  -lm `glib-config --libs`
+CFLAGS += $(CPPFLAGS) -g3 -c -pthread  -DUNIX  -I../base/ -I../dynlibsrc `pkg-config --cflags glib-2.0`
+LDFLAGS += -g -L../base/ -L../dynlibsrc  -lm `pkg-config --libs glib-2.0`
 AR_OPTIONS = ru
 
 RANLIB_NEEDED = 0
--- wise-2.4.1.orig/src/base/makefile
+++ wise-2.4.1/src/base/makefile
@@ -19,9 +19,9 @@
 
 
 
-#CFLAGS = -c  -g -DUNIX -DWISE_MEMORY_WATCH
-CFLAGS = -c  -O2 -DUNIX -pthread
-LFLAGS =  -g  -lm 
+#CFLAGS += -c  -g -DUNIX -DWISE_MEMORY_WATCH
+CFLAGS += $(CPPFLAGS) -c -DUNIX -pthread
+LDFLAGS += -g -lm 
 DFLAGS = -n Wise2_ -a _api.h -b _api.t -l -D
 AR_OPTIONS = ru
 
--- wise-2.4.1.orig/src/HMMer2/Makefile
+++ wise-2.4.1/src/HMMer2/Makefile
@@ -13,7 +13,7 @@
 ## your compiler and compiler flags
 #
 CC     = cc
-CFLAGS = -g 
+CFLAGS += $(CPPFLAGS) -g 
 
 ## machine specific definitions
 #
@@ -75,25 +75,25 @@
 all: 	$(PROGS)
 
 hmmalign: $(OBJ) hmmalign.o
-	$(CC) $(CFLAGS) $(MDEFS) -o $@ hmmalign.o $(OBJ) $(LIBS) 
+	$(CC) $(CFLAGS) $(LDFLAGS) $(MDEFS) -o $@ hmmalign.o $(OBJ) $(LIBS) 
 
 hmmbuild: $(OBJ) hmmbuild.o
-	$(CC) $(CFLAGS) $(MDEFS) -o $@ hmmbuild.o $(OBJ) $(LIBS) 
+	$(CC) $(CFLAGS) $(LDFLAGS) $(MDEFS) -o $@ hmmbuild.o $(OBJ) $(LIBS) 
 
 hmmcalibrate: $(OBJ) hmmcalibrate.o
-	$(CC) $(CFLAGS) $(MDEFS) -o $@ hmmcalibrate.o $(OBJ) $(LIBS) 
+	$(CC) $(CFLAGS) $(LDFLAGS) $(MDEFS) -o $@ hmmcalibrate.o $(OBJ) $(LIBS) 
 
 hmmconvert: $(OBJ) hmmconvert.o
-	$(CC) $(CFLAGS) $(MDEFS) -o $@ hmmconvert.o $(OBJ) $(LIBS) 
+	$(CC) $(CFLAGS) $(LDFLAGS) $(MDEFS) -o $@ hmmconvert.o $(OBJ) $(LIBS) 
 
 hmmemit: $(OBJ) hmmemit.o
-	$(CC) $(CFLAGS) $(MDEFS) -o $@ hmmemit.o $(OBJ) $(LIBS) 
+	$(CC) $(CFLAGS) $(LDFLAGS) $(MDEFS) -o $@ hmmemit.o $(OBJ) $(LIBS) 
 
 hmmpfam: $(OBJ) hmmpfam.o
-	$(CC) $(CFLAGS) $(MDEFS) -o $@ hmmpfam.o $(OBJ) $(LIBS) 
+	$(CC) $(CFLAGS) $(LDFLAGS) $(MDEFS) -o $@ hmmpfam.o $(OBJ) $(LIBS) 
 
 hmmsearch: $(OBJ) hmmsearch.o
-	$(CC) $(CFLAGS) $(MDEFS) -o $@ hmmsearch.o $(OBJ) $(LIBS) 
+	$(CC) $(CFLAGS) $(LDFLAGS) $(MDEFS) -o $@ hmmsearch.o $(OBJ) $(LIBS) 
 
 #######
 ## Target used in making Shiva test suite
